File: backend/Krios/krios/Models/Staff.cs
namespace Krios.Models.Krios
{
    public class Staff
    {
        public long id { get; set; }
        public long orgId { get; set; }
        public long locationId { get; set; }
        public string name { get; set; } = "";
        public string role { get; set; } = "";
        public string outlet { get; set; } = "";
        public string phone { get; set; } = "";
        public string email { get; set; } = "";
        public DateTime? joinDate { get; set; }
        public long baseSalary { get; set; }
        public decimal commissionRate { get; set; }
        public long target { get; set; }
        public string status { get; set; } = "";
        public string address { get; set; } = "";
        public string bankName { get; set; } = "";
        public string bankAccount { get; set; } = "";
        public string bankIfsc { get; set; } = "";
        public string idProofType { get; set; } = "";
        public string idProofRef { get; set; } = "";
        public string addressProofType { get; set; } = "";
        public string addressProofRef { get; set; } = "";
        public string createdby { get; set; } = "";
        public DateTime? createdon { get; set; }
        public string updatedby { get; set; } = "";
        public DateTime? updatedon { get; set; }
    }

    public class StaffSelectReq
    {
        public long id { get; set; }
        public long orgId { get; set; }
        public long locationId { get; set; }
        public string search { get; set; } = "";
        public string status { get; set; } = "";
    }

    public class StaffDeleteReq
    {
        public long id { get; set; }
        public long orgId { get; set; }
    }
}

File: frontend/src/model/staff.ts
/** Models for table: staff (ids are number / BIGINT) */

export class StaffRes {
  id: number = 0;
  orgId: number = 0;
  locationId: number = 0;
  name: string = "";
  role: string = "";
  outlet: string = "";
  phone: string = "";
  email: string = "";
  joinDate: string = "";
  baseSalary: number = 0;
  commissionRate: number = 0;
  target: number = 0;
  status: string = "";
  address: string = "";
  bankName: string = "";
  bankAccount: string = "";
  bankIfsc: string = "";
  idProofType: string = "";
  idProofRef: string = "";
  addressProofType: string = "";
  addressProofRef: string = "";
  createdby: string = "";
  createdon: string = "";
  updatedby: string = "";
  updatedon: string = "";
}

export class CreateStaffReq {
  orgId: number = 0;
  locationId: number = 0;
  name: string = "";
  role: string = "";
  outlet: string = "";
  phone: string = "";
  email: string = "";
  joinDate: string = "";
  baseSalary: number = 0;
  commissionRate: number = 0;
  target: number = 0;
  status: string = "";
  address: string = "";
  bankName: string = "";
  bankAccount: string = "";
  bankIfsc: string = "";
  idProofType: string = "";
  idProofRef: string = "";
  addressProofType: string = "";
  addressProofRef: string = "";
}

export class CreateStaffRes extends StaffRes {}

export class UpdateStaffReq {
  id: number = 0;
  orgId: number = 0;
  locationId: number = 0;
  name: string = "";
  role: string = "";
  outlet: string = "";
  phone: string = "";
  email: string = "";
  joinDate: string = "";
  baseSalary: number = 0;
  commissionRate: number = 0;
  target: number = 0;
  status: string = "";
  address: string = "";
  bankName: string = "";
  bankAccount: string = "";
  bankIfsc: string = "";
  idProofType: string = "";
  idProofRef: string = "";
  addressProofType: string = "";
  addressProofRef: string = "";
}

export class UpdateStaffRes extends StaffRes {}

export class GetStaffReq {
  id: number = 0;
  orgId: number = 0;
}

export class GetStaffRes extends StaffRes {
  errorMessage: string = "";
}

export class ListStaffReq {
  orgId: number = 0;
  locationId: number = 0;
  search: string = "";
  page: number = 1;
  pageSize: number = 50;
}

export class ListStaffRes {
  items: StaffRes[] = [];
  total: number = 0;
  errorMessage: string = "";
}

export class DeleteStaffReq {
  id: number = 0;
  orgId: number = 0;
}

export class DeleteStaffRes {
  id: number = 0;
  success: boolean = false;
  errorMessage: string = "";
}
